### PR TITLE
ci: consolidate with github actions

### PR DESCRIPTION
**Pull request by xoviat**
remove travis, appveyor

### CODE REVIEW

File: .travis.yml
Comment (bwoodsend) [inline .travis.yml:27]: This appears to have gotten lost?

File: appveyor.yml
Comment (bwoodsend) [inline appveyor.yml:131]: Hmm, wonder if we still need to explicitly enable long paths on Github actions...

File: .github/workflows/windows.yml
Comment (bwoodsend) [inline .github/workflows/windows.yml]: That hard-coded path is not the same on Github. Try:
```suggestion
      - name: Get pip cache dir
        run: echo "pip_cache_dir=$(pip cache dir)" >> $GITHUB_ENV
        shell: bash
        
      - name: cache
        id: cache-target
        uses: actions/cache@v2
        with:
          path: ${{ env.pip_cache_dir }}
          key: ${{ runner.os }}-${{ matrix.python-version }}
```

**Review by bwoodsend — CHANGES_REQUESTED** [review]

**Comment by bwoodsend** [comment]
Thanks for working on this BTW...

File: .github/workflows/windows.yml
Comment (xoviat) [reply .github/workflows/windows.yml:75]: I'm the one who put it in.

File: .github/workflows/windows.yml
Comment (bwoodsend) [reply .github/workflows/windows.yml]: Ahh I see you've just looked them up.

File: .github/workflows/windows.yml
Comment (xoviat) [reply .github/workflows/windows.yml]: It shows the tests on appveyor (and with this, GitHub).

**Comment by xoviat** [comment]
@bwoodsend Anything else?

File: .github/workflows/windows.yml
Comment (bwoodsend) [reply .github/workflows/windows.yml]: I can see them anyway though. What's the difference? It appears to work identically on the [hooks repo](https://github.com/pyinstaller/pyinstaller-hooks-contrib/) which doesn't have junit.

Comment (rokm) [reply .github/workflows/windows.yml]: There's a message about insufficient permissions in the [workflow view](https://github.com/pyinstaller/pyinstaller/actions/runs/720243844).

File: appveyor.yml
Comment (bwoodsend) [reply appveyor.yml:131]: [I see](https://github.com/pyinstaller/pyinstaller/runs/2271880377?check_suite_focus=true#step:12:973) this is no longer a problem.

**Comment by bwoodsend** [comment]
No I think that's it from me.

**Review by bwoodsend — APPROVED** [review]

**Review by rokm — APPROVED** [review]
Looks good to me, as well.

Except we can probably do without the "Publish Test Report" part, as it is not working due to insufficient permissions.

File: .github/workflows/windows.yml
Comment (bwoodsend) [reply .github/workflows/windows.yml]: The permissions are probably insufficient because this is a fork. I suppose that means it can't run on pull requests because they will most likely be forks too?

**Comment by xoviat** [comment]
example of failure with the junit: https://github.com/xoviat/pyinstaller/actions/runs/720599599

**Comment by bwoodsend** [comment]
Cool, we'll leave it in then as it is more readable than checking each log to find where each one failed. I'll merge this tomorrow once its last CI job is done.

**Comment by Legorooj** [comment]
I'd like to hold off on merging this just yet.

File: .github/workflows/ci.yml
Comment (Legorooj) [inline .github/workflows/ci.yml:11]: Can we exclude `pyup/**` and `master` here?

```suggestion
      - '**'
      - '!master'
      - '!pyup/**'
```

File: .github/workflows/ci.yml
Comment (Legorooj) [inline .github/workflows/ci.yml:72]: We don't need `distclean` to be explicitly specified; `all` includes it.

Comment (Legorooj) [inline .github/workflows/ci.yml:72]: `distclean` is included as part of `all`.
```suggestion
          python waf all
```

**Review by Legorooj — CHANGES_REQUESTED** [review]

**Comment by xoviat** [comment]
Note: the new pytest GH actions plugin that I created directly displays the test failure on the file: https://github.com/xoviat/pyinstaller/pull/2/files

**Comment by bwoodsend** [comment]
Hmm, totally overkill but I like it anyway.

File: .github/workflows/ci.yml
Comment (bwoodsend) [inline .github/workflows/ci.yml:91]: Can we keep these as soft `startsWith()` conditions?
```suggestion
        if: ${{ !startsWith(matrix.os, 'macos') }}
```
The reason being that occasionally we like to try specific OS versions like `macos-11.0` which go wrong here. Same with the next one down.

**Review by bwoodsend — CHANGES_REQUESTED** [review]

**Review by bwoodsend — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (Legorooj) [inline .github/workflows/ci.yml:105]:
```suggestion
        if: startsWith(matrix.os, 'macos')
```

**Review by Legorooj — CHANGES_REQUESTED** [review]

**Review by Legorooj — APPROVED** [review]

**Pull request merged**
